result/array-bracket-spacing.js

```diff
@@ -0,0 +1,118 @@
+/*eslint-env es6*/
+
+var arr = [ 'foo', 'bar' ];
+var [ x, y ] = z;
+
+var arr = ['foo', 'bar'];
+var [x,y] = z;
+/*eslint array-bracket-spacing: ["error", "never"]*/
+/*eslint-env es6*/
+
+var arr = [ 'foo', 'bar' ];
+var arr = ['foo', 'bar' ];
+var arr = [ ['foo'], 'bar'];
+var arr = [[ 'foo' ], 'bar'];
+var arr = [ 'foo',
+  'bar'
+];
+var [ x, y ] = z;
+var [ x,y ] = z;
+var [ x, ...y ] = z;
+var [ ,,x, ] = z;
+/*eslint array-bracket-spacing: ["error", "never"]*/
+/*eslint-env es6*/
+
+var arr = [];
+var arr = ['foo', 'bar', 'baz'];
+var arr = [['foo'], 'bar', 'baz'];
+var arr = [
+  'foo',
+  'bar',
+  'baz'
+];
+var arr = ['foo',
+  'bar'
+];
+var arr = [
+  'foo',
+  'bar'];
+
+var [x, y] = z;
+var [x,y] = z;
+var [x, ...y] = z;
+var [,,x,] = z;
+/*eslint array-bracket-spacing: ["error", "always"]*/
+/*eslint-env es6*/
+
+var arr = ['foo', 'bar'];
+var arr = ['foo', 'bar' ];
+var arr = [ ['foo'], 'bar' ];
+var arr = ['foo',
+  'bar'
+];
+var arr = [
+  'foo',
+  'bar'];
+
+var [x, y] = z;
+var [x,y] = z;
+var [x, ...y] = z;
+var [,,x,] = z;
+/*eslint array-bracket-spacing: ["error", "always"]*/
+/*eslint-env es6*/
+
+var arr = [];
+var arr = [ 'foo', 'bar', 'baz' ];
+var arr = [ [ 'foo' ], 'bar', 'baz' ];
+var arr = [ 'foo',
+  'bar'
+];
+var arr = [
+  'foo',
+  'bar' ];
+var arr = [
+  'foo',
+  'bar',
+  'baz'
+];
+
+var [ x, y ] = z;
+var [ x,y ] = z;
+var [ x, ...y ] = z;
+var [ ,,x, ] = z;
+/*eslint array-bracket-spacing: ["error", "always", { "singleValue": false }]*/
+
+var foo = [ 'foo' ];
+var foo = [ 'foo'];
+var foo = ['foo' ];
+var foo = [ 1 ];
+var foo = [ 1];
+var foo = [1 ];
+var foo = [ [ 1, 2 ] ];
+var foo = [ { 'foo': 'bar' } ];
+/*eslint array-bracket-spacing: ["error", "always", { "singleValue": false }]*/
+
+var foo = ['foo'];
+var foo = [1];
```

Comment:
__[development][eslint]__
* Error - 'foo' is already defined (no-redeclare)
* Error - A space is required after '[' (array-bracket-spacing)
* Error - A space is required before ']' (array-bracket-spacing)